python/paddle/v2/framework/optimizer.py

```diff
@@ -108,6 +109,26 @@ def _get_accumulator(self, name, param):
                             format(name, param.name))
         return self._accumulators[name][param.name]
 
+    def _increment_global_step(self, block):
+        """Increment the global step by 1 after every iteration
+
+        Args:
+            block: the block in which the loss variable is present
+
+        Returns:
+            list with global_step increment op as its only element
+        """
+        assert isinstance(block, framework.Block)
+        assert self._global_step is not None
+        # create the increment op
+        increment_op = block.append_op(
+            type="increment",
+            inputs={"X": self._global_step},
+            outputs={"Out": self._global_step},
+            attrs={"step": 1.0})
+
+        return [increment_op]
```

Comment:
why return a list

Reply:
Yes you are right. I should change this to an op

Reply:
🤝

Reply:
Fixed in a747012f47d8fc30b4b53de364bfb818ced104dc